Automattic/simplenote-ios
# Add Checklists

This PR adds supports for 'checkable' checklists in the note editor. There's a new button in the toolbar that will insert a new list:

<img width="419" alt="screen shot 2018-12-12 at 9 50 57 am" src="https://user-images.githubusercontent.com/789137/49906908-c9a54a80-fe28-11e8-9e3f-1d6b07171628.png">

Behind the scenes, the note saves with markdown checklist syntax (e.g. `- [ ], - [x]`). The note editor now uses a regex to parse which lines start with checklist markdown, and replaces them with tappable `NSTextAttachment` objects in the attributedText of the editor UITextView. See the `attributedStringWithChecklistAttachments` method.

When a note is saved, the new method `getPlainTextContent` searches for the `NSTextAttachments`, and replaces them with their plain text counterpart.

The checklists are also rendered in the note list.

Known issue:
- [x] The checkboxes aren't tapple on iOS 10. (making this require >= iOS 11)

**To Test**
* Open a note, and create a checklist! Pressing `return` at the end of a checklist item should automatically add a new one on the next line. The keyboard should also be capitalized after this action (I had to add some iffy workarounds for this 😬).
* Open the note on app.simplenote.com, it should have the correct markdown syntax displayed.
* Update the note on the web browser, the app should update and display any modified checklists appropriately.
* Test things that modify the note content, like the history browser and ensure they work as expected.
* Run the new tests!
